tracer/src/Datadog.Trace/Util/ProcessHelpers.cs

```diff
@@ -70,11 +71,107 @@ public static void GetCurrentProcessInformation(out string processName, out stri
         public static void GetCurrentProcessRuntimeMetrics(out TimeSpan userProcessorTime, out TimeSpan systemCpuTime, out int threadCount, out long privateMemorySize)
         {
             using var process = Process.GetCurrentProcess();
-
             userProcessorTime = process.UserProcessorTime;
             systemCpuTime = process.PrivilegedProcessorTime;
             threadCount = process.Threads.Count;
             privateMemorySize = process.PrivateMemorySize64;
         }
+
+        /// <summary>
+        /// Run a command and get the standard output content as a string
+        /// </summary>
+        /// <param name="command">Command to run</param>
+        /// <param name="input">Standard input content</param>
+        /// <returns>Task with the content of the standard output</returns>
+        public static async Task<string> RunCommandAsync(Command command, string input = null)
+        {
+            Log.Debug("Running command: {command} {args}", command.Cmd, command.Arguments);
+            var processStartInfo = GetProcessStartInfo(command);
+            if (input is not null)
+            {
+                processStartInfo.RedirectStandardInput = true;
+            }
+
+            using var processInfo = Process.Start(processStartInfo);
+            if (processInfo is null)
+            {
+                return null;
+            }
+
+            if (input is not null)
+            {
+                await processInfo.StandardInput.WriteAsync(input).ConfigureAwait(false);
+                await processInfo.StandardInput.FlushAsync().ConfigureAwait(false);
+                processInfo.StandardInput.Close();
+            }
+
+            string output = null;
+            while (!processInfo.HasExited)
+            {
+                output += await processInfo.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+                await Task.Delay(1).ConfigureAwait(false);
```

Comment:
I was thinking to change it to `Task.Delay(15)` instead (because basically is the minimum time of a delay in .net?? / https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?redirectedfrom=MSDN&view=net-6.0#System_Threading_Tasks_Task_Delay_System_Int32_), the idea is to give time to the command to flush data to stdout before we try to read the stream again. I'm worried that `Task.Yield()` would be too fast and I want to avoid a close loop consuming cpu on nothing.

Reply:
Changed to `Task.Delay(15)` in https://github.com/DataDog/dd-trace-dotnet/pull/3045/commits/ee3048f6cb4f6c1027a372cdedc58883b5193527